File: snackdown2017/QualificationRound/2ndRound/snakeCoup.py
def fence(a,b,n):
	#bool hori = a.index('*')>=0 and b.index('*')>=0

	if a.index('*')>=0 and b.index('*')>=0:
		f=int(1)
	else:
		f=int(0)	

	prod = [False,False]
	for i in range(n):
		for k in range(0,2):
			if a[i]=='*' or b[i]=='*':
				if prod[k]:
					f+=1
					prod[0]==prod[1]==False
					break
		for k in range(0,2):
			if a[k]=='*' and b[k]=='*':
				prod[k]=True
	return f


for _ in range(int(raw_input())):

	n = int(raw_input())
	a = str(raw_input())
	b = str(raw_input())

	print fence(a,b,n)

File: snackdown2017/QualificationRound/2ndRound/SnakeCoup.java
import java.util.*;
class SnakeCoup {
    
    static int fences2(String[] matrix, int n)
    {
        boolean hori=matrix[0].indexOf('*')>=0&&matrix[1].indexOf('*')>=0;     
        System.out.println(hori);
        int f=hori?1:0;
        System.out.println(f);
        boolean prod[]=new boolean[]{false,false};
        for (int i=0; i<n; i++)  {
            for (int k=0; k<2;k++) {
                if (matrix[k].charAt(i)=='*') {
                    if (prod[k]) {
                        f++;
                        prod[0]=prod[1]=false;
                        break;
                    }
                }
            }
            for (int k=0; k<2;k++)
                if (matrix[k].charAt(i)=='*')
                    prod[k]=true; 
        }
        return f;
    }
    
    static Scanner sc = new Scanner(System.in);
    public static void main(String[] args)
    {      
        int T=sc.nextInt();     
        for (int i=0; i<T; i++) {
            int n=sc.nextInt(); 
            String[] matrix=new String[2];
            matrix[0]=sc.next();
            matrix[1]=sc.next();
            System.out.println(fences2(matrix,n));
    }
}
}